From 791509067bf5acdfb92247c14ee33a9a628346e6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 16 Aug 2025 17:26:05 +0800
Subject: [PATCH] thread_pool.h: avoid ambiguous size types in Config struct

Config must have exact size, but bool is not guaranteed to be 1 byte. Use uint8_t instead,
which is always 1 byte. Fixes the build on powerpc-darwin.

Fixes: https://github.com/google/highway/issues/2679
---
 hwy/contrib/thread_pool/thread_pool.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git hwy/contrib/thread_pool/thread_pool.h hwy/contrib/thread_pool/thread_pool.h
index 2afac04c..bc0eb47e 100644
--- hwy/contrib/thread_pool/thread_pool.h
+++ hwy/contrib/thread_pool/thread_pool.h
@@ -272,7 +272,7 @@ class Config {  // 8 bytes
   SpinType spin_type;
   WaitType wait_type;
   BarrierType barrier_type;
-  bool exit;
+  uint8_t exit;
   uint32_t reserved = 0;
 };
 static_assert(sizeof(Config) == 8, "");
